core/src/main/java/software/amazon/awssdk/core/auth/ProfileCredentialsProvider.java

```diff
@@ -15,48 +15,65 @@
- * SDK for Java and the AWS CLI.</p>
+ * Credentials provider based on AWS configuration profiles. This loads credentials from a {@link ProfileFile}, allowing you to
+ * share multiple sets of AWS security credentials between different tools like the AWS SDK for Java and the AWS CLI.
  *
  * <p>See http://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html</p>
  *
- * @see ProfilesConfigFile
+ * <p>If this credentials provider is loading assume-role credentials from STS, it should be cleaned up with {@link #close()} if
+ * it is no longer being used.</p>
+ *
+ * @see ProfileFile
  */
-public class ProfileCredentialsProvider extends FileSystemCredentialsProvider {
-    private final ProfilesConfigFile profilesConfigFile;
+public final class ProfileCredentialsProvider implements AwsCredentialsProvider, SdkAutoCloseable {
+    private final AwsCredentialsProvider profileCredentialsProvider;
+    private final RuntimeException loadException;
+
+    private final ProfileFile profileFile;
     private final String profileName;
 
     /**
      * @see #builder()
      */
     private ProfileCredentialsProvider(Builder builder) {
-        if (builder.profilesConfigFile == null) {
-            Validate.notNull(builder.profileFileLocationProvider, "Profile file location provider must not be null.");
-            File defaultProfileFile = builder.profileFileLocationProvider.getLocation();
-            this.profilesConfigFile = defaultProfileFile == null ? null : ProfilesConfigFile.create(defaultProfileFile);
+        this.profileName = builder.profileName != null ? builder.profileName
+                                                       : AwsSystemSetting.AWS_PROFILE.getStringValueOrThrow();
+
+        // Load the profiles file
+        this.profileFile = Optional.ofNullable(builder.profileFile)
+                                   .orElseGet(builder.defaultProfileFileLoader);
+
+        // Load the profile and credentials provider
+        this.profileCredentialsProvider = profileFile.profile(profileName)
+                                                     .flatMap(Profile::credentialsProvider)
+                                                     .orElse(null);
+
+        // If we couldn't load the credentials provider for some reason, save an exception describing why. This exception will
+        // only be raised on calls to getCredentials. We don't want to raise an exception here because it may be expected (eg. in
+        // the default credential chain).
+        if (profileCredentialsProvider == null) {
+            String loadError = String.format("Profile file contained no credentials for profile '%s': %s",
+                                             profileName, profileFile);
+            this.loadException = new SdkClientException(loadError);
         } else {
-            this.profilesConfigFile = builder.profilesConfigFile;
+            this.loadException = null;
```

Comment:
do we need this `else` block? isn't it default to be null?

Reply:
By default it's not initialized (final), so we have to do it here.